### DISCUSSION TITLE
Spinel Frame size set to 1300

### DISCUSSION THREAD

**Discussion by jchellap** [openthread/openthread#9306] [General]
In the spinel source code for open thread, the spinel frame size is set to 1300.
**#define SPINEL_FRAME_MAX_SIZE 1300**
I read in the spinel core as 1300 is the recommended size and is capable of handling from 2 to 1300 bytes.
We have a memory constrained device, hence if we reduce the size from 1300 to a smaller number say 200 or 300, what is the impact of the thread standard performance?
The device is expected to work as SSED node, RCP design

**Comment by zhanglongxia** [comment]
There are two parts that may use the long Spinel frame. One is the log frame (SPINEL_PROP_STREAM_DEBUG and SPINEL_PROP_STREAM_LOG), the other one is the diag cli message (SPINEL_PROP_NEST_STREAM_MFG). If you set the size to 200 or 300, these two kinds of Spinel messages may be truncated. 
The SPINEL_FRAME_MAX_SIZE  is used to limit the max length of the Spinel frame, I don't think it will affect the Thread performance.

**Reply by jchellap** [reply]
Thanks for the clarification. What is the max length required to support this? I mean using 500 etc will cater or 1300 is the size to be used if the frame should not be truncated? What is the safe size for a low end node can use?

**Reply by jchellap** [reply]
@zhanglongxia , I am just getting knowhow on the implementation. 
One is the log frame (SPINEL_PROP_STREAM_DEBUG and SPINEL_PROP_STREAM_LOG), the other one is the diag cli message (SPINEL_PROP_NEST_STREAM_MFG)
Would like to know if these are enabled by default? Also, where can i find the spinel log? Some documentation or guidance will help understand this part better.

**Comment by zhanglongxia** [comment]
> 



> @zhanglongxia , I am just getting knowhow on the implementation. One is the log frame (SPINEL_PROP_STREAM_DEBUG and SPINEL_PROP_STREAM_LOG), the other one is the diag cli message (SPINEL_PROP_NEST_STREAM_MFG) Would like to know if these are enabled by default? Also, where can i find the spinel log? Some documentation or guidance will help understand this part better.

@jchellap For the log messages, the `OPENTHREAD_CONFIG_LOG_LEVEL` is set `OT_LOG_LEVEL_CRIT` by default.  You can set the `OPENTHREAD_CONFIG_LOG_LEVEL` to `OT_LOG_LEVEL_NONE` in your project to disable the log messages. The log messages from RCP are written to the [host log system](https://github.com/openthread/openthread/blob/main/src/lib/spinel/radio_spinel_impl.hpp#L758) with the prefix `RCP => `, you can find the log message in the file `/var/log/syslog` on linux.  There are no docs for this, you need to check the code.

For the cli diag messages, it is enabled by the configuration `OPENTHREAD_CONFIG_DIAG_ENABLE`, and it is set to [0](https://github.com/openthread/openthread/blob/main/src/core/config/diag.h#L45) by default. The cli output messages can be found in the [radio commands](https://github.com/openthread/openthread/blob/main/src/core/diags/factory_diags.cpp#L72) . The cli output messages are not too long, I think 200 bytes should be ok.

**Reply by jchellap** [reply]
Thanks @zhanglongxia for the clarification
